File: app/src/main/java/com/ternaryop/utils/MimeType.java
package com.ternaryop.utils;

import java.util.HashMap;

import android.webkit.MimeTypeMap;

/**
 * Created by dave on 27/04/14.
 * Helper class to work with Mime types
 */
public class MimeType {
    public static MimeType instance;
    private final HashMap<String, String> map;

    private MimeType() {
        map = new HashMap<>();
        map.put("flv", "video/x-flv");
        map.put("mp4", "video/mp4");
        map.put("m3u8", "application/x-mpegURL");
        map.put("ts", "video/MP2T");
        map.put("3gp", "video/3gpp");
        map.put("mov", "video/quicktime");
        map.put("avi", "video/x-msvideo");
        map.put("wmv", "video/x-ms-wmv");
    }

    public static MimeType getInstance() {
        if (instance == null) {
            instance = new MimeType();
        }
        return instance;
    }

    public String getTypeFromMimeType(String mimeType) {
        String[] comps = mimeType.split("/");
        if (comps.length == 2) {
            return comps[0];
        }
        return mimeType;
    }

    public static boolean isVideoMimeType(String mimeType) {
        return mimeType != null && mimeType.startsWith("video/");
    }

    public static boolean isImageMimeType(String mimeType) {
        return mimeType != null && mimeType.startsWith("image/");
    }

    public String getMimeTypeFromExtension(String extension) {
        String mimeType = map.get(extension);

        if (mimeType == null) {
            mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
        }
        return mimeType;
    }
}

File: app/src/main/java/com/ternaryop/utils/MimeType.kt
package com.ternaryop.utils

import android.webkit.MimeTypeMap

/**
 * Created by dave on 27/04/14.
 * Helper class to work with Mime types
 */
object MimeType {
    private val map = mapOf(
        "flv" to "video/x-flv",
        "mp4" to "video/mp4",
        "m3u8" to "application/x-mpegURL",
        "ts" to "video/MP2T",
        "3gp" to "video/3gpp",
        "mov" to "video/quicktime",
        "avi" to "video/x-msvideo",
        "wmv" to "video/x-ms-wmv"
    )

    fun getTypeFromMimeType(mimeType: String): String {
        val comps = mimeType.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return if (comps.size == 2) {
            comps[0]
        } else mimeType
    }

    fun getMimeTypeFromExtension(extension: String): String {
        return map[extension] ?: return MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension)
    }

    fun isVideoMimeType(mimeType: String): Boolean = mimeType.startsWith("video/")
    fun isImageMimeType(mimeType: String): Boolean = mimeType.startsWith("image/")
}
